Branch-SDK/src/io/branch/referral/RemoteInterface.java

```diff
@@ -431,4 +456,18 @@ private String convertJSONtoString(JSONObject json) {
 
         return result.toString();
     }
+
+
+    private String getResolvedApiUrl( String url) throws MalformedURLException, UnknownHostException {
+        // If the hostname resolves to multiple IP addresses, URL connection will attempt to connect the next if one connection fails
+        // This may result is multiple timeouts elapse before the connect attempt throws an exception. In order to make sure the
+        // Branch callback is called with in the timeout specified we resolve the host first and then try only with first resolved ip
+
+        String apiHost = new URL(url).getHost();
+        InetAddress inetAddress = InetAddress.getByName(apiHost);
```

Comment:
This has been tested and since the cached entries expires after TTL this seems not an issue.  Also    existing connections honored until TTL in anycase.